haskell/src/Monocle/Api/Server.hs

```diff
@@ -299,6 +309,141 @@ crawlerCommitInfo request = do
         . Enumerated
         $ Right err
 
+-- | /task_data endpoints
+data TDError = TDUnknownApiKey | TDUnknownCrawler | TDUnknownIndex | TDDateInvalid deriving (Show)
+
+validateTaskDataRequest ::
+  -- | The index name
+  LText ->
+  -- | The crawler name
+  LText ->
+  -- | The api key
+  Maybe LText ->
+  -- | True if the timestamp must be checked
+  Bool ->
+  -- | the commit timestamp
+  Maybe T.Timestamp ->
```

Comment:
It seems like we could check the timestamp when it is set. Also it is better to avoid boolean blindness.

```suggestion
  -- | The commit timestamp
  Maybe T.Timestamp ->
```